modmail/extensions/configuration_manager.py

```diff
@@ -0,0 +1,221 @@
+
+    return options
+
+
+class KeyConverter(commands.Converter):
+    """Convert argument into a configuration key."""
+
+    async def convert(self, ctx: Context, arg: str) -> KeyT:
+        """Ensure that a key is of the valid format, allowing a user to input other formats."""
+        # basically we're converting an argument to a key.
+        # config keys are delimited by `.`, and always lowercase, which means that we can make a few passes
+        # before actually trying to make any guesses.
+        # the problems are twofold. a: we want to suggest other keys
+        # and b: the interface needs to be easy to use.
+
+        # as a partial solution for this, `/`, `-`, `.` are all valid delimiters and are converted to `.`
+        # depending on common problems, it is *possible* to add `_` but would require fuzzy matching over
+        # all of the keys since that can also be a valid character name.
+
+        fields = get_all_conf_options(config.default().__class__)
+
+        new_arg = ""
+        for c in arg.lower():
+            if c in " /`":
+                new_arg += "."
+            else:
+                new_arg += c
+
+        if new_arg in fields:
+            return new_arg
+        else:
+            raise commands.BadArgument(
+                f"{ctx.current_parameter.name} {arg} is not a valid configuration key."
+            )
+
+
+class ConfigurationManager(ModmailCog, name="Configuration Manager"):
+    """Manage the bot configuration."""
+
+    config_fields: typing.Dict[str, ConfOptions]
+
+    def __init__(self, bot: ModmailBot):
+        self.bot = bot
+
+        self.config_fields = get_all_conf_options(config.default().__class__)
+
+    @commands.group(name="config", aliases=("cfg", "conf"), invoke_without_command=True)
+    async def config_group(self, ctx: Context) -> None:
+        """Manage the bot configuration."""
+        if ctx.invoked_subcommand is None:
+            await ctx.send_help(ctx.command)
+
+    @config_group.command(name="list")
+    async def list_config(self, ctx: Context) -> None:
+        """List the valid configuration options."""
+        options = {}
+        for table, opt in self.config_fields.items():
+            if opt.hidden or opt.frozen:
+                continue
+
+            # we want to merge items from the same config table so they are on the same table
+            key = table.rsplit(".", 1)[0]
+            options[key] = options.get(key, "") + (
+                "\n".join(
+                    [
+                        f"**{string.capwords(opt.canonical_name or opt.name)}**",
+                        f"Default: `{opt.default}`"
+                        if opt.default is not None
+                        else "Required. There is no default value for this option.",
+                        f"{opt.description}",
+                        f" {opt.extended_description}\n" if opt.extended_description else "",
+                    ]
+                )
+            )
```

Comment:
Let's use a temporary variable here rather than use indentation and wrapping to this degree, it's difficult and confusing to read. 

What about a multiline f string with """?